presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/TransformCorrelatedSingleRowSubqueryToProject.java

```diff
@@ -72,13 +72,28 @@ public Result apply(LateralJoinNode parent, Captures captures, Context context)
                 .where(ProjectNode.class::isInstance)
                 .findAll();
 
-        if (subqueryProjections.size() == 0) {
+        // LateralJoinNode's input's output symbols will be added as in the projections
+        // during building the result ProjectNode. However, InlineProjections might have
+        // added the same projection already, so only keep one of them
```

Comment:
Can you please explain me what projections do you see here. I would love to know why `InlineProjections` and `RemoveRedundantIdentityProjections` do not work here.

Reply:
@kokosing 

https://github.com/prestodb/presto/pull/11121#discussion_r205733808